CDP4CrossViewEditor/Generator/CrossviewSheetPMeanUtilities.cs

```diff
@@ -0,0 +1,235 @@
+        /// </summary>
+        public const string PStandBy = "P_stby";
+
+        /// <summary>
+        /// Power peak parameter type short name
+        /// </summary>
+        private const string PPeak = "P_peak";
+
+        /// <summary>
+        /// Power duty cycle parameter type short name
+        /// </summary>
+        public const string PDutyCycle = "P_duty_cyc";
+
+        /// <summary>
+        /// Power mean parameter type short name
+        /// </summary>
+        public static string PMean = "P_mean";
+
+        /// <summary>
+        /// Hardcoded power related parameter type short names list
+        /// </summary>
+        public static readonly string[] PowerParameters = { PRedundancy, POn, PStandBy, PPeak, PDutyCycle, PMean };
+
+        /// <summary>
+        /// Hardcoded redundancy.scheme
+        /// NOTE: not sure about this value
+        /// </summary>
+        private const string RedundancySchemeHot = "Active";
+
+        /// <summary>
+        /// Hardcoded redundancy.scheme
+        /// NOTE: not sure about this value
+        /// </summary>
+        private const string RedundancySchemeCold = "Passive";
+
+        /// <summary>
+        /// Hardcoded redundancy.type
+        /// </summary>
+        private const string RedundancyTypeInternal = "Internal";
+
+        /// <summary>
+        /// Hardcoded redundancy.type
+        /// </summary>
+        private const string RedundancyTypeExternal = "External";
+
+        /// <summary>
+        /// Check if P_mean parameter calculation is possible
+        /// </summary>
+        /// <param name="parameters">
+        /// List of the parameters <see cref="List{ParameterOrOverrideBase}"/>
+        /// </param>
+        /// <returns>
+        /// True if all power related parameters are present, False otherwise
+        /// </returns>
+        public static bool IsCalculationPossible(List<ParameterOrOverrideBase> parameters)
+        {
+            // Check if all power required parameters are present
+            if (PowerParameters.Select(p => p.ToLowerInvariant()).Except(new[] { "P_peak".ToLowerInvariant() })
+                .Except(parameters.Select(p => p.ParameterType.ShortName.ToLowerInvariant())).Any())
+            {
+                return false;
+            }
+
+            var pDutyCycle = parameters.Where(p => p != null)
+                .FirstOrDefault(p => p.ParameterType.ShortName.Equals("P_duty_cyc", StringComparison.InvariantCultureIgnoreCase));
+
+            var pMean = parameters.Where(p => p != null)
+                .FirstOrDefault(p => p.ParameterType.ShortName.Equals("P_mean", StringComparison.InvariantCultureIgnoreCase));
+
+            if (pDutyCycle == null || pMean == null)
+            {
+                return false;
+            }
+
+            // Check if P_duty_cyc && P_mean have the same option dependency
+            if (pDutyCycle.IsOptionDependent != pMean.IsOptionDependent)
```

Comment:
Option dependency needs to be taken into account for all parameters, not just "p_duty_cyc" and "p_mean".

Reply:
Fixed as of `aae9b12`.